src/node/build.ts

```diff
@@ -76,6 +76,9 @@ export async function build(ssgOptions: Partial<ViteSSGOptions> = {}, viteConfig
       },
     },
     mode: config.mode,
+    define: {
+      'import.meta.env.SSR': JSON.stringify(undefined),
```

Comment:
we should check also old Vite versions, and augment vite import meta at least here (or use `process.env.VITE_SSG`)